### ISSUE TITLE
Significant wait between games on fast GPU.

### ISSUE THREAD

**Issue by jjoshua2** [glinscott/leela-chess#132]
With a 1080ti the moves happen very fast and a game might take 30 seconds, and then there is a decent time of about 10 seconds with no gpu utilization between games while it is zipping, checking for a new network, and uploading the training.0.gz file. 

Running 3 at once seems to mitigate this, and utilization normally stays at 80%, but on windows still requires copying 3x folders, and 3x the server load checking and dl networks.

I think the relevant lines in the go client are the train and upload, and the upload could be made async? 

Download latest network could also run async, with the risk of doing 1 extra game with the old network, and having to store both networks probably in case the new one doesn't finish downloading in time.

I believe the gzip happens in the lczero and not the client, and thus could not be easily async without moving the gzip to the client. But this is probably much faster than network delays.

```
	trainFile, pgn := train(networkPath)
	uploadGame(httpClient, trainFile, pgn, nextGame)
```

I think I read that go automatically does things async if it sees there is nothing that requires it to be done in order? I'm not familiar with go language to know if there is async http versions.

**Comment by Uriopass** [comment]
About the multiples folders on windows it's a known issue and should be solved soon, and about the upload I think uploading asynchronously shouldn't be a big problem, since after the folder issue is resolved there wouldn't be a problem of ending a new game after an old one is still uploading. 
I will try to make a PR to update the client tomorrow, as the main contributors are busy on the training and the matching.

**Comment by Uriopass** [comment]
@jjoshua2 
https://groups.google.com/forum/#!topic/lczero/QMYkUrswnl4

There you go

**Comment by killerducky** [comment]
@Uriopass thanks for fixing the issue. But I think we should wait for the pull request to be reviewed and merge before distributing binaries. This one is probably ok but there might be something unexpected that we find in the review process. I think it's best for people to use the officially released binaries.

**Comment by Uriopass** [comment]
@killerducky Yeah you're right there might be a bug or a problem I haven't seen. I'll remove the binaries and wait for glinscott to make them.

**Comment by glinscott** [comment]
@Uriopass thanks for putting this together!  I will test it out tomorrow.

**Comment by jjoshua2** [comment]
I think this is done

**Comment by Uriopass** [comment]
It should be fixed by #225 , but gary is afraid that it doesn't properly cleans everything.
